### ISSUE TITLE
[Feature Request] RDP Auto Discover

### ISSUE THREAD

**Issue by IngwiePhoenix** [xpipe-io/xpipe#739]
Hello!

I just found XPipe by sheer chance and this might be the management tool I have been looking for. The current solutions used in my company are not very compatible with my accessibility needs - but XPipe, is. So I want to put it through the ringer.

In particular, our network is a whole lot of Windows hosts and I would like to auto-detect the RDP connection and add it to it's own collection - much like the SSH method, but for RDP instead.

Because entering ~30 hosts by hand is tiring and kinda slow. :)

Thank you for this amazing tool and kind regards,

Ingwie

**Comment by crschnick** [comment]
Auto-detect how exactly? Do you mean the network scan but for RDP instead of SSH?

Also in terms of accessibility, feel free to report anything that doesn't work as intended. Because it's hard to know for myself if I don't use it with accessibility features all the time

**Comment by IngwiePhoenix** [comment]
> Auto-detect how exactly? Do you mean the network scan but for RDP instead of SSH?

Yes, precisely. Think `nmap -p 3389 <ip cidr>`, effectively.

> Also in terms of accessibility, feel free to report anything that doesn't work as intended

You already do like 90% right. Some buttons do not have hover labels so I have no idea what they do - but the UI has dark and light themes, you use native tools (Windows Terminal, native SSH and RDP clients) instead of embedding everything inside - which means that I can pick and chose tools that work best for me. Prime example: Windows' built-in RDP client is the only one that actually, properly handles my screen magnifier. Because when I use ctrl+alt+mousewheel, that is handled locally instead of being forwarded. Most clients end up nomming those keystrokes and sending them to the remote, making my magnifier gloriously unusable. So, so far, I have no complaints.

But the more I use it, the more I will come across stuff and let you know, for sure! =)

**Comment by crschnick** [comment]
Alright, I think that can be added.

About the accessibility, feel free to try out https://github.com/xpipe-io/xpipe-ptb. In the latest release I tried to fix all accessibility issues that I could find

**Comment by crschnick** [comment]
This scan feature is also now implemented in the PTB releases

The check whether a port is open is suprisingly difficult from a windows system due to limited cli tools there. But I hope it works well enough

**Comment by crschnick** [comment]
This is now released
